test/extended/openstack/machine.go

```diff
@@ -64,18 +70,36 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] Machine", func()
 	})
 
 	g.It("ProviderSpec is correctly applied to OpenStack instances", func() {
+		type ServerWithAZ struct {
+			servers.Server
+			availabilityzones.ServerAvailabilityZoneExt
+		}
 		for _, machine := range machineResources {
+			var err error
+			var instance ServerWithAZ
+			var machineNetworks []string
+			var machineNetwork *networks.Network
 			machineName := machine.Get("metadata.name").String()
 			machineFlavor := machine.Get("spec.providerSpec.value.flavor").String()
 			machineImage := machine.Get("spec.providerSpec.value.image").String()
+			machineAZ := machine.Get("spec.providerSpec.value.availabilityZone").String()
+			machineRootVolCinderAz := machine.Get("spec.providerSpec.value.rootVolume.availabilityZone").String()
+			machineRootVolVolumeType := machine.Get("spec.providerSpec.value.rootVolume.volumeType").String()
 			machineSecurityGroups := make(map[string]struct{})
+			for _, network := range objects(machine.Get("spec.providerSpec.value.networks")) {
+				if network.Get("subnets").String() == "" {
```

Comment:
why the machinenetwork name is only being retrieved when subnet is not defined?

Reply:
To get only the additional networks